Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Favorites/AssetBrowserFavoritesModel.cpp

```diff
@@ -0,0 +1,377 @@
+        {
+            if (action == Qt::IgnoreAction)
+                return true;
+
+            AZStd::vector<const AssetBrowserEntry*> entries;
+
+            if (Utils::FromMimeData(data, entries))
+            {
+                for (auto entry : entries)
+                {
+                    if (!entry)
+                    {
+                        continue;
+                    }
+                    if (entry->RTTI_IsTypeOf(SourceAssetBrowserEntry::RTTI_Type()) ||
+                        entry->RTTI_IsTypeOf(ProductAssetBrowserEntry::RTTI_Type()) ||
+                        entry->RTTI_IsTypeOf(FolderAssetBrowserEntry::RTTI_Type()))
+                    {
+                        AssetBrowserFavoriteRequestBus::Broadcast(&AssetBrowserFavoriteRequestBus::Events::AddFavoriteAsset, entry);
+                    }
+                }
+                return true;
+            }
+
+            return QAbstractItemModel::dropMimeData(data, action, row, column, parent);
+        }
+
+        Qt::DropActions AssetBrowserFavoritesModel::supportedDropActions() const
+        {
+            return Qt::CopyAction;
+        }
+
+        QMimeData* AssetBrowserFavoritesModel::mimeData(const QModelIndexList& indexes) const
+        {
+            QMimeData* mimeData = new QMimeData;
+
+            AZStd::vector<const AssetBrowserEntry*> collected;
+            collected.reserve(indexes.size());
+
+            for (const auto& index : indexes)
+            {
+                if (index.isValid())
+                {
+                    const AssetBrowserFavoriteItem* item = static_cast<const AssetBrowserFavoriteItem*>(index.internalPointer());
+                    if (item->GetFavoriteType() == AssetBrowserFavoriteItem::FavoriteType::AssetBrowserEntry)
+                    {
+                        const EntryAssetBrowserFavoriteItem* favoriteItem = static_cast<const EntryAssetBrowserFavoriteItem*>(item);
+                        const AssetBrowserEntry* entry = favoriteItem->GetEntry();
+                        if (entry->GetEntryType() != AssetBrowserEntry::AssetEntryType::Product ||
+                            entry->GetEntryType() != AssetBrowserEntry::AssetEntryType::Source)
+                        {
+                            collected.push_back(entry);
+                        }
+                    }
+                }
+            }
+
+            Utils::ToMimeData(mimeData, collected);
+
+            return mimeData;
+        }
+
+        void AssetBrowserFavoritesModel::SetSearchWidget(SearchWidget* searchWidget)
+        {
+            m_searchWidget = searchWidget;
+
+            LoadFavorites();
+        }
+
+        void AssetBrowserFavoritesModel::FavoritesChanged()
+        {
+            if (m_modifying)
+            {
+                return;
+            }
+
+            LoadFavorites();
+        }
+
+        void AssetBrowserFavoritesModel::Select(AssetBrowserFavoriteItem* favoriteItem)
```

Comment:
perhaps this should also be triggered when a user double clicks a search/asset

Reply:
Done